hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanHelper.groovy

```diff
@@ -0,0 +1,234 @@
+    String newline = System.getProperty("line.separator");
+    return HtmlUtil.toText(response).replaceAll("azkaban.failure.message=", newline + newline);
+  }
+
+  /**
+   * Reads the session id from session.file in the user's ~/.azkaban directory.
+   *
+   * @return sessionId The saved session id
+   */
+  static String readSession() {
+    logger.lifecycle("AzkabanHelper: readSession");
+    File file = new File(System.getProperty("user.home") + "/.azkaban/session.properties");
+    String sessionId = null;
+    if (file.exists()) {
+      file.withInputStream { inputStream ->
+        Properties properties = new Properties();
+        properties.load(inputStream);
+        sessionId = properties.getProperty("sessionId");
+      }
+    }
+    return sessionId;
+  }
+
+  /**
+   * Stores the session id in a properties file under ~/.azkaban.
+   *
+   * @param sessionId The session id
+   */
+  static void saveSession(String sessionId) {
+    logger.lifecycle("AzkabanHelper: saveSession");
+    if (!sessionId) {
+      throw new GradleException("No session ID obtained to save");
+    }
+
+    File dir = new File(System.getProperty("user.home") + "/.azkaban");
+    if (!dir.exists() && !dir.mkdirs()) {
+      logger.error("Unable to create directory: " + dir.toString());
+      return;
+    }
+
+    File file = new File(dir, "session.properties");
+    if (file.exists() && !file.delete()) {
+      logger.error("Unable to delete the existing file at: " + file.toString());
+      return;
+    }
+
+    try {
+      file.withWriter { writer ->
+        Properties properties = new Properties();
+        properties.setProperty("sessionId", sessionId);
+        properties.store(writer, null);
+      }
+
+      // Make the file readable only by the user. The Java File API makes this a little awkward to express.
+      file.setReadable(false, false);
+      file.setReadable(true, true);
+    }
+    catch (IOException ex) {
+      logger.error("Unable to store session ID to file: " + file.toString() + "\n" + ex.toString());
+    }
+  }
+
+  /**
+   * Makes sure that the pre-requisites of using the azkaban tasks are satisfied.
+   *
+   * @param azkProject The AzkabanProject
+   */
+  static void configureTask(AzkabanProject azkProject) {
+    def console = System.console();
+    if (console == null) {
+      throw new GradleException(
+          "\nCannot access the system console. To use this task, explicitly set JAVA_HOME to the version specified in product-spec.json (at LinkedIn) and pass --no-daemon in your command.");
+    }
+
+    if (!azkProject.azkabanUserName) {
+      azkProject.azkabanUserName = console.readLine("\nAzkaban user name: ");
+      logger.lifecycle("To avoid entering username: Refer https://iwww.corp.linkedin.com/wiki/cf/display/GRID/Hadoop+Gradle+Plugin+User+Guide#HadoopGradlePluginUserGuide-ConfigureAzkabanUploadInformation");
+      //why don't we write it to .azkabanPlugin.json after upload is successuful
```

Comment:
Can you remove these two "why don't we" comments? I have some ideas on how to automate the .azkabanPlugin.json file.